### PR TITLE
gobgp: init at 2.23.0

### PR DESCRIPTION
**Pull request by higebu**
###### Motivation for this change

A CLI tool for GoBGP. GoBGP is an open source BGP implementation. https://github.com/osrg/gobgp
I will create a gobgpd package as a next step.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
    - 15MB
    ```console
    $ nix path-info -S $(which gobgp)
    /nix/store/mbmrqik607diwsdls8i78fn8j4pjjsbr-gobgp-2.23.0           15041048
    ```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10526,4 +10526,10 @@
     github = "jbcrail";
     githubId = 6038;
   };
+  higebu = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:10529]: Please put this into a separate commit with the message ``maintainers: add higebu`` and sort it alphabetically.

File: pkgs/tools/networking/gobgp/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  vendorSha256 = "0dmd4r6x76jn8pyvp47x4llzc2wij5m9lchgyaagcb5sfdgbns9x";
+
+  postConfigure = ''
+    export CGO_ENABLED=0
+  '';
+
+  buildFlagsArray = ''
+    -ldflags=
+    -s -extldflags '-static'
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/gobgp/default.nix:22]:
```suggestion
    -s -w -extldflags '-static'
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by higebu** [comment]
@SuperSandro2000 Thank you for your review. I fixed your requests.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111083` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gobgp</li>
</ul>
</details>
